bin/test-storybook.js

```diff
@@ -34,32 +40,65 @@ function sanitizeURL(url) {
   return finalURL;
 }
 
-const targetURL = sanitizeURL(process.env.TARGET_URL || `http://localhost:6006`);
-
-urlExists(targetURL, function (err, exists) {
-  if (!exists) {
-    console.error(`[test-storybook] It seems that your Storybook instance is not running at: ${targetURL}. Are you sure it's running?`)
-    process.exit(1)
-  }
-
-  executeJestPlaywright()
-});
-
-function executeJestPlaywright() {
-  const fs = require('fs');
-  const path = require('path');
-
+async function executeJestPlaywright(args) {
   const jest = require('jest');
-  let argv = process.argv.slice(2);
+  let argv = args.slice(2);
 
   const jestConfigPath = fs.existsSync('test-runner-jest.config.js')
     ? 'test-runner-jest.config.js'
-    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js')
+    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js');
 
-  argv.push(
-    '--config',
-    jestConfigPath
-  );
+  argv.push('--config', jestConfigPath);
 
-  jest.run(argv);
+  await jest.run(argv);
+}
+
+async function checkStorybook(url) {
+  try {
+    const res = await fetch(url, { method: 'HEAD' });
+    if (res.status !== 200) throw new Error(`Unxpected status: ${res.status}`);
+  } catch (e) {
+    console.error(
+      `[test-storybook] It seems that your Storybook instance is not running at: ${url}. Are you sure it's running?`
+    );
+    process.exit(1);
+  }
 }
+
+async function fetchStoriesJson(url) {
+  const storiesJsonUrl = new URL('/stories.json', url).toString();
+  let tmpFile;
+  try {
+    const res = await fetch(storiesJsonUrl);
+    const tmpDir = tempy.directory();
+    tmpFile = path.join(tmpDir, 'stories-json.test.js');
+    const json = await res.text();
+    const js = transformPlaywrightJson(json);
+    fs.writeFileSync(tmpFile, js);
+  } catch (err) {
+    console.error(`[test-storybook] Failed to fetch stories.json from ${storiesJsonUrl}.`);
+    process.exit(1);
+  }
+  return tmpFile;
+}
+
+const main = async () => {
+  const targetURL = sanitizeURL(process.env.TARGET_URL || `http://localhost:6006`);
+  await checkStorybook(targetURL);
+  let args = process.argv.filter((arg) => arg !== '--stories-json');
+  let testFile;
+  if (args.length !== process.argv.length) {
+    testFile = await fetchStoriesJson(targetURL);
+    process.env.TEST_ROOT = path.dirname(testFile);
+    process.env.TEST_MATCH = '**/*.test.js';
+  }
+
+  await executeJestPlaywright(args);
+
+  if (testFile) {
```

Comment:
This won't happen if the tests are cancelled with cmd + C and the files will never be deleted:

![image](https://user-images.githubusercontent.com/1671563/151350348-128b6090-6a2f-4494-add9-3f4a778b233e.png)